### PR TITLE
glew: Fix glu and glx dependencies

### PR DESCRIPTION
**Pull request by chuckatkins**

### CODE REVIEW

File: var/spack/repos/builtin/packages/glew/package.py [hunk 1]
```diff
@@ -9,18 +9,24 @@ class Glew(CMakePackage):
...
-    root_cmakelists_dir = "build/cmake"
+    # glu is already forcibly disabled in the CMakeLists.txt.  This prevents
+    # it from showing up in the .pc file
+    patch('remove-pkgconfig-glu-dep.patch')
+
+    def cmake_args(self):
+        args = [
+            self.define('BUILD_UTILS', True),
+            self.define('GLEW_REGAL', False),
+            self.define('GLEW_OSMESA', False),
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/glew/package.py:29]: Is it outside the scope of this to allow an osmesa option?

Comment (chuckatkins) [reply var/spack/repos/builtin/packages/glew/package.py:29]: I did that at first but the logic is more complicated than it should be and necessitates a fair amount of cleanup to GLEW's `CMakeLists.txt`.

**Review by kwryankrattiger — APPROVED** [review]
+2 Changes look good as is, tested on linux box.

File: var/spack/repos/builtin/packages/glew/package.py (hunk 1 above)
Comment (kwryankrattiger) [reply var/spack/repos/builtin/packages/glew/package.py:29]: ah, too bad.

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256.

**Pull request merged**